eng/common/TestResources/setup-az-modules.yml

```diff
@@ -0,0 +1,36 @@
+# Cloud Configuration will be splat into parameters of `Add-AzEnvironment`. It
+# should be JSON in the form (not all fields are required):
+# {
+#   "Name": "<environment name>",
+#   "PublishSettingsFileUrl": "<publish settings file url>",
+#   "ServiceEndpoint": "<service endpoint>",
+#   "ManagementPortalUrl": "<management portal url>",
+#   "ActiveDirectoryEndpoint": "<active directory endpoint>",
+#   "ActiveDirectoryServiceEndpointResourceId": "<active directory service endpoint resource id>",
+#   "ResourceManagerEndpoint": "<resource manager endpoint>",
+#   "GalleryEndpoint": "<gallery endpoint>",
+#   "GraphEndpoint": "<graph endpoint>",
+#   "GraphAudience": "<graph audience>",
+#   "AzureKeyVaultDnsSuffix": "<key vault suffix>",
+#   "AzureKeyVaultServiceEndpointResourceId": "<key vault service endpoint resource id>"
+# }
+
+steps:
+  - bash: sudo chown -R runner ~/.Azure
+    displayName: (MacOS) Grant access to ~/.Azure
+    condition: contains(variables['OSVmImage'], 'mac')
+
+  # New-TestResources command requires Az module
+  - pwsh: Install-Module -Name Az -Scope CurrentUser -AllowClobber -Force -Verbose
+    displayName: Install Azure PowerShell module
+
+  - task: Powershell@2
```

Comment:
Why not use the `pwsh` task like you did above? More concise and consistent.

Reply:
For inline scripts like this yes I agree using pwsh works well. For one-line script calls we generally try to use the Powershell task directly as it is easier to diagnose and get the exact command line for debugging and trying to run it locally. FYI I've written up our guidance around this at https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/2/Azure-DevOps-Pipeline-Guidance?anchor=powershell-tips